openwisp_radius/tests/__init__.py

```diff
@@ -0,0 +1,127 @@
+import os
+
+from django.core.management import call_command
+
+from ..base.models import _encode_secret
+
+# it's 21 of April on UTC, this date is fabricated on purpose
+# to test possible timezone related bugs in the date filtering
+_TEST_DATE = '2019-04-20T22:14:09-04:00'
+
+
+class CreateRadiusObjectsMixin(object):
+    def _get_defaults(self, opts, model=None):
+        options = {}
+        options.update(opts)
+        return options
+
+    def _create_radius_check(self, **kwargs):
+        if kwargs.get('value'):
+            kwargs['value'] = _encode_secret(kwargs['attribute'], kwargs.get('value'))
+        options = self._get_defaults(kwargs)
+        rc = self.radius_check_model(**options)
+        rc.full_clean()
+        rc.save()
+        return rc
+
+    def _create_radius_accounting(self, **kwargs):
+        options = self._get_defaults(kwargs)
+        ra = self.radius_accounting_model(**options)
+        ra.full_clean()
+        ra.save()
+        return ra
+
+    def _create_radius_reply(self, **kwargs):
+        options = self._get_defaults(kwargs)
+        rr = self.radius_reply_model(**options)
+        rr.full_clean()
+        rr.save()
+        return rr
+
+    def _create_nas(self, **kwargs):
+        options = self._get_defaults(kwargs)
+        n = self.nas_model(**options)
+        n.full_clean()
+        n.save()
+        return n
+
+    def _create_radius_group(self, **kwargs):
+        options = self._get_defaults(kwargs)
+        rg = self.radius_group_model(**options)
+        rg.full_clean()
+        rg.save()
+        return rg
+
+    def _create_radius_groupcheck(self, **kwargs):
+        options = self._get_defaults(kwargs, model=self.radius_groupcheck_model)
+        c = self.radius_groupcheck_model(**options)
+        c.full_clean()
+        c.save()
+        return c
+
+    def _create_radius_groupreply(self, **kwargs):
+        options = self._get_defaults(kwargs, model=self.radius_groupreply_model)
+        r = self.radius_groupreply_model(**options)
+        r.full_clean()
+        r.save()
+        return r
+
+    def _create_radius_usergroup(self, **kwargs):
+        options = self._get_defaults(kwargs, model=self.radius_usergroup_model)
+        ug = self.radius_usergroup_model(**options)
+        ug.full_clean()
+        ug.save()
+        return ug
+
+    def _create_radius_postauth(self, **kwargs):
+        options = self._get_defaults(kwargs)
+        rp = self.radius_postauth_model(**options)
```

Comment:
we're not required to do this anymore, we can call the model loaded with swappable directly if you don't see counterindications (we're already doing this elsewhere).

A string and replace on the `tests` directory would be enough to replace all occurrences.